From 4c60444a35303a243cd3b35a9d72e69afb54d809 Mon Sep 17 00:00:00 2001
From: Ernestas Kulik <ekulik@redhat.com>
Date: Wed, 10 Jun 2020 08:32:39 +0200
Subject: [PATCH 5/8] Adapt more test ouput to Java 11

Paths to runtime classes are no longer printed out, so we can simplify
the driver magic.
---
 test/outputs/run_test.log.in | 108 ++++++++++++++++++-----------------
 test/testdriver              |   7 +--
 2 files changed, 56 insertions(+), 59 deletions(-)

diff --git a/test/outputs/run_test.log.in b/test/outputs/run_test.log.in
index aa2b110..ae4a894 100644
--- a/test/outputs/run_test.log.in
+++ b/test/outputs/run_test.log.in
@@ -1,98 +1,99 @@
 Caught exception java.io.FileNotFoundException in method java.io.FileInputStream.open0()
 Exception in thread "main" java.io.FileNotFoundException: _wrong_file_ (No such file or directory)
-	at java.io.FileInputStream.open0(Native Method) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileInputStream.class]
-	at java.io.FileInputStream.open(FileInputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileInputStream.class]
-	at java.io.FileInputStream.<init>(FileInputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileInputStream.class]
+	at java.base/java.io.FileInputStream.open0(Native Method) [jrt:/java.base/java/io/FileInputStream.class]
+	at java.base/java.io.FileInputStream.open(FileInputStream.java:LINENO) [jrt:/java.base/java/io/FileInputStream.class]
+	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:LINENO) [jrt:/java.base/java/io/FileInputStream.class]
 	at Test.readWrongFile(Test.java:89) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.fileRelatedIssues(Test.java:461) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:513) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.io.FileNotFoundException in method java.io.FileInputStream.open0()
 Exception in thread "main" java.io.FileNotFoundException: /root/.bashrc (Permission denied)
-	at java.io.FileInputStream.open0(Native Method) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileInputStream.class]
-	at java.io.FileInputStream.open(FileInputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileInputStream.class]
-	at java.io.FileInputStream.<init>(FileInputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileInputStream.class]
+	at java.base/java.io.FileInputStream.open0(Native Method) [jrt:/java.base/java/io/FileInputStream.class]
+	at java.base/java.io.FileInputStream.open(FileInputStream.java:LINENO) [jrt:/java.base/java/io/FileInputStream.class]
+	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:LINENO) [jrt:/java.base/java/io/FileInputStream.class]
 	at Test.readUnreadableFile(Test.java:111) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.fileRelatedIssues(Test.java:462) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:513) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.io.FileNotFoundException in method java.io.FileOutputStream.open0()
 Exception in thread "main" java.io.FileNotFoundException: /root/.bashrc (Permission denied)
-	at java.io.FileOutputStream.open0(Native Method) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileOutputStream.class]
-	at java.io.FileOutputStream.open(FileOutputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileOutputStream.class]
-	at java.io.FileOutputStream.<init>(FileOutputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileOutputStream.class]
-	at java.io.FileOutputStream.<init>(FileOutputStream.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/io/FileOutputStream.class]
+	at java.base/java.io.FileOutputStream.open0(Native Method) [jrt:/java.base/java/io/FileOutputStream.class]
+	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:LINENO) [jrt:/java.base/java/io/FileOutputStream.class]
+	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:LINENO) [jrt:/java.base/java/io/FileOutputStream.class]
+	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:LINENO) [jrt:/java.base/java/io/FileOutputStream.class]
 	at Test.writeToUnwritableFile(Test.java:134) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.fileRelatedIssues(Test.java:463) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:513) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.net.UnknownHostException in method java.net.Inet4AddressImpl.lookupAllHostAddr()
 Exception in thread "main" java.net.UnknownHostException: xyzzy: Name or service not known
-	at java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Inet4AddressImpl.class]
-	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetAddress$2.class]
-	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetAddress.class]
-	at java.net.InetAddress.getAllByName0(InetAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetAddress.class]
-	at java.net.InetAddress.getAllByName(InetAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetAddress.class]
-	at java.net.InetAddress.getAllByName(InetAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetAddress.class]
-	at java.net.InetAddress.getByName(InetAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetAddress.class]
-	at java.net.InetSocketAddress.<init>(InetSocketAddress.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/InetSocketAddress.class]
-	at java.net.Socket.<init>(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
+	at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method) [jrt:/java.base/java/net/Inet4AddressImpl.class]
+	at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress$PlatformNameService.class]
+	at java.base/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress.class]
+	at java.base/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress$NameServiceAddresses.class]
+	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress.class]
+	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress.class]
+	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress.class]
+	at java.base/java.net.InetAddress.getByName(InetAddress.java:LINENO) [jrt:/java.base/java/net/InetAddress.class]
+	at java.base/java.net.InetSocketAddress.<init>(InetSocketAddress.java:LINENO) [jrt:/java.base/java/net/InetSocketAddress.class]
+	at java.base/java.net.Socket.<init>(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
 	at Test.readFromUnknownHost(Test.java:157) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.networkRelatedIssues(Test.java:473) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:514) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.net.UnknownHostException in method java.net.AbstractPlainSocketImpl.connect()
 Exception in thread "main" java.net.UnknownHostException: xyzzy
-	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/AbstractPlainSocketImpl.class]
-	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/SocksSocketImpl.class]
-	at java.net.Socket.connect(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
-	at java.net.Socket.connect(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
-	at java.net.Socket.<init>(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
-	at java.net.Socket.<init>(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
+	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:LINENO) [jrt:/java.base/java/net/AbstractPlainSocketImpl.class]
+	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:LINENO) [jrt:/java.base/java/net/SocksSocketImpl.class]
+	at java.base/java.net.Socket.connect(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
+	at java.base/java.net.Socket.connect(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
+	at java.base/java.net.Socket.<init>(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
+	at java.base/java.net.Socket.<init>(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
 	at Test.readFromUnknownHost(Test.java:157) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.networkRelatedIssues(Test.java:473) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:514) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.net.ConnectException in method java.net.PlainSocketImpl.socketConnect()
 Exception in thread "main" java.net.ConnectException: Connection refused (Connection refused)
-	at java.net.PlainSocketImpl.socketConnect(Native Method) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/PlainSocketImpl.class]
-	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/AbstractPlainSocketImpl.class]
-	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/AbstractPlainSocketImpl.class]
-	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/AbstractPlainSocketImpl.class]
-	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/SocksSocketImpl.class]
-	at java.net.Socket.connect(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
-	at java.net.Socket.connect(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
-	at java.net.Socket.<init>(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
-	at java.net.Socket.<init>(Socket.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/Socket.class]
+	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method) [jrt:/java.base/java/net/PlainSocketImpl.class]
+	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:LINENO) [jrt:/java.base/java/net/AbstractPlainSocketImpl.class]
+	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:LINENO) [jrt:/java.base/java/net/AbstractPlainSocketImpl.class]
+	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:LINENO) [jrt:/java.base/java/net/AbstractPlainSocketImpl.class]
+	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:LINENO) [jrt:/java.base/java/net/SocksSocketImpl.class]
+	at java.base/java.net.Socket.connect(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
+	at java.base/java.net.Socket.connect(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
+	at java.base/java.net.Socket.<init>(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
+	at java.base/java.net.Socket.<init>(Socket.java:LINENO) [jrt:/java.base/java/net/Socket.class]
 	at Test.readFromSocket(Test.java:177) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.networkRelatedIssues(Test.java:474) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:514) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.lang.NumberFormatException in method java.lang.Long.parseLong()
 Exception in thread "main" java.lang.NumberFormatException: null
-	at java.lang.Long.parseLong(Long.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/Long.class]
-	at java.lang.Long.parseLong(Long.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/Long.class]
-	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/sun/net/www/protocol/http/HttpURLConnection.class]
-	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/sun/net/www/protocol/http/HttpURLConnection.class]
-	at java.net.URL.openStream(URL.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/URL.class]
+	at java.base/java.lang.Long.parseLong(Long.java:LINENO) [jrt:/java.base/java/lang/Long.class]
+	at java.base/java.lang.Long.parseLong(Long.java:LINENO) [jrt:/java.base/java/lang/Long.class]
+	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
+	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
+	at java.base/java.net.URL.openStream(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
 	at Test.readFromURL(Test.java:238) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.networkRelatedIssues(Test.java:475) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:514) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.io.FileNotFoundException in method sun.net.www.protocol.http.HttpURLConnection.getInputStream0()
 Exception in thread "main" java.io.FileNotFoundException: http://localhost:54321/_this_does_not_exists_
-	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/sun/net/www/protocol/http/HttpURLConnection.class]
-	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/sun/net/www/protocol/http/HttpURLConnection.class]
-	at java.net.URL.openStream(URL.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/URL.class]
+	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
+	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:LINENO) [jrt:/java.base/sun/net/www/protocol/http/HttpURLConnection.class]
+	at java.base/java.net.URL.openStream(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
 	at Test.readFromURL(Test.java:238) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.networkRelatedIssues(Test.java:475) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:514) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.net.MalformedURLException in method java.net.URL.<init>()
 Exception in thread "main" java.net.MalformedURLException: no protocol: @#$%^&malformed URL@#$%^&*()
-	at java.net.URL.<init>(URL.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/URL.class]
-	at java.net.URL.<init>(URL.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/URL.class]
-	at java.net.URL.<init>(URL.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/net/URL.class]
+	at java.base/java.net.URL.<init>(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
+	at java.base/java.net.URL.<init>(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
+	at java.base/java.net.URL.<init>(URL.java:LINENO) [jrt:/java.base/java/net/URL.class]
 	at Test.malformedURL(Test.java:194) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.networkRelatedIssues(Test.java:476) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:514) [file:@CMAKE_BINARY_DIR@/test/Test.class]
@@ -104,9 +105,10 @@ Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 42 ou
 	at Test.throwAndCatchAllExceptions(Test.java:486) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.main(Test.java:515) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
-Caught exception java.lang.StringIndexOutOfBoundsException in method java.lang.String.charAt()
+Caught exception java.lang.StringIndexOutOfBoundsException in method java.lang.StringLatin1.charAt()
 Exception in thread "main" java.lang.StringIndexOutOfBoundsException: String index out of range: -1
-	at java.lang.String.charAt(String.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/String.class]
+	at java.base/java.lang.StringLatin1.charAt(StringLatin1.java:LINENO) [jrt:/java.base/java/lang/StringLatin1.class]
+	at java.base/java.lang.String.charAt(String.java:LINENO) [jrt:/java.base/java/lang/String.class]
 	at Test.throwStringIndexOutOfBoundsException(Test.java:278) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.catchStringIndexOutOfBoundsException(Test.java:362) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.throwAndCatchAllExceptions(Test.java:487) [file:@CMAKE_BINARY_DIR@/test/Test.class]
@@ -120,7 +122,7 @@ Exception in thread "main" java.lang.NullPointerException
 	at Test.main(Test.java:515) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.lang.ClassCastException in method Test.throwClassCastException()
-Exception in thread "main" java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
+Exception in thread "main" java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
 	at Test.throwClassCastException(Test.java:302) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.catchClassCastException(Test.java:392) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.throwAndCatchAllExceptions(Test.java:489) [file:@CMAKE_BINARY_DIR@/test/Test.class]
@@ -128,8 +130,8 @@ Exception in thread "main" java.lang.ClassCastException: java.lang.Integer canno
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.lang.UnsatisfiedLinkError in method java.lang.Runtime.load0()
 Exception in thread "main" java.lang.UnsatisfiedLinkError: Expecting an absolute path of the library: xyzzy
-	at java.lang.Runtime.load0(Runtime.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/Runtime.class]
-	at java.lang.System.load(System.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/System.class]
+	at java.base/java.lang.Runtime.load0(Runtime.java:LINENO) [jrt:/java.base/java/lang/Runtime.class]
+	at java.base/java.lang.System.load(System.java:LINENO) [jrt:/java.base/java/lang/System.class]
 	at Test.loadLibrary(Test.java:325) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.catchUnsatisfiedLinkErrorUserLibrary(Test.java:422) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.throwAndCatchAllExceptions(Test.java:491) [file:@CMAKE_BINARY_DIR@/test/Test.class]
@@ -137,9 +139,9 @@ Exception in thread "main" java.lang.UnsatisfiedLinkError: Expecting an absolute
 executable: @CMAKE_BINARY_DIR@/test/Test.class
 Caught exception java.lang.UnsatisfiedLinkError in method java.lang.ClassLoader.loadLibrary()
 Exception in thread "main" java.lang.UnsatisfiedLinkError: no xyzzy in java.library.path
-	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/ClassLoader.class]
-	at java.lang.Runtime.loadLibrary0(Runtime.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/Runtime.class]
-	at java.lang.System.loadLibrary(System.java:LINENO) [jar:file:JAVA_AND_SYSTEM_SPECIFIC_PATH/rt.jar!/java/lang/System.class]
+	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:LINENO) [jrt:/java.base/java/lang/ClassLoader.class]
+	at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:LINENO) [jrt:/java.base/java/lang/Runtime.class]
+	at java.base/java.lang.System.loadLibrary(System.java:LINENO) [jrt:/java.base/java/lang/System.class]
 	at Test.loadSystemLibrary(Test.java:336) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.catchUnsatisfiedLinkErrorSystemLibrary(Test.java:437) [file:@CMAKE_BINARY_DIR@/test/Test.class]
 	at Test.throwAndCatchAllExceptions(Test.java:492) [file:@CMAKE_BINARY_DIR@/test/Test.class]
diff --git a/test/testdriver b/test/testdriver
index b02bc5c..ff71685 100644
--- a/test/testdriver
+++ b/test/testdriver
@@ -29,12 +29,7 @@ else
     cp $4 $TMP_RESULT
 fi
 
-JAVA_RTLIB_PATH=`sed -n "s/^.*\(jar:file:\)\(.*\)\(\/rt.jar\!.*\).*$/\2/p" $TMP_RESULT | head -1`
-
-if [ -n "$JAVA_RTLIB_PATH" ]; then
-    sed -i "s,"$JAVA_RTLIB_PATH",JAVA_AND_SYSTEM_SPECIFIC_PATH," $TMP_RESULT
-    sed -i "s/\(\.java:\)[0-9]\+\() \[.*JAVA_AND_SYSTEM_SPECIFIC_PATH\)/\1LINENO\2/" $TMP_RESULT
-fi
+sed -i "s/\(\.java:\)[0-9]\+\() \[jrt:.*\]\)/\1LINENO\2/" $TMP_RESULT
 
 diff -u $3 $TMP_RESULT
 EC=$?
-- 
2.26.2

